pages/index.tsx

```diff
@@ -38,6 +40,32 @@ function KeyboardComponent() {
   const [input, setInput] = useState("");
   const [layout, setLayout] = useState("default");
   const keyboard = useRef();
+  let pressedKeysArr = [];
+  let currentlyPressedKeys = Array.from(useKeyTracker().isPressed).join(' ')
+
+
+  
+  // TODO: handleShift and Caps Lock via physical keyboard. The following only handles digital keyboard
+  const handleShift = () => {
+    setLayout(layout === "default" ? "shift" : "default")
+  }
+
+  const onKeyPress = (button) => {
+    console.log("Button pressed", button);
+    /**
+     * If you want to handle the shift and caps lock buttons
+     */
+    if (button === "{shift}" || button === "{lock}") handleShift();
+  };
+
+
+  const keyTracker = useKeyTracker()
```

Comment:
nit: `useKeyTracker` is called  for a second time here, which means that the key state tracking is happening twice under the hood. It would be best to only call useKeyTracker once at the top of the component and reuse the returned instance.

Reply:
Changes made